main/bamboo/bamboo.go

```diff
@@ -171,6 +180,67 @@ func listenToZookeeper(conf configuration.Configuration, eventBus *event_bus.Eve
 	return serviceConn
 }
 
+func listenToEventStream(conf configuration.Configuration, eventBus *event_bus.EventBus) {
+	client := &http.Client{}
+	client.Timeout = 0 * time.Second
+
+	for _, marathon := range conf.Marathon.Endpoints() {
+		go func() {
+			for {
+				req, err := http.NewRequest("GET", marathon+"/v2/events", nil)
+				req.Header.Set("Accept", "text/event-stream")
+				if err != nil {
+					errorMsg := "An error occurred while creating request to Marathon events system: %s\n"
+					log.Printf(errorMsg, err)
+					continue
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					continue
```

Comment:
Let me describe whole scheme:

1) try to open connection
2) if failed, wait 1 sec and go to 1)
3) if not, read data from connection
4) if failed, log and go to 1 without delay
5) if not, read event and go to 3)


Which means that:
If failure happened while reading event on open connection, then whole cycle will start from 1)
If failure happened while connection opening then each attempt will happen not more than once per second.

So:
It's ok to wait 1 second if it's impossible to open connection to marathon.
If error happened during "read" phase, then we should retry connection immediately because it could be just dropped connection, but Marathon is still able to serve requests. If it's not, then it will loop on 1)-2).